external-crates/move/documentation/book/src/packages.md

```diff
@@ -91,30 +151,114 @@ specified_address = "0xB0B"
 # Local dependency
 LocalDep = { local = "projects/move-awesomeness", addr_subst = { "std" = "0x1" } }
 # Git dependency
-MoveStdlib = { git = "https://github.com/diem/diem.git", subdir="language/move-stdlib", rev = "56ab033cc403b489e891424a629e76f643d4fb6b" }
+MoveStdlib = { git = "https://github.com/MystenLabs/sui.git", subdir = "crates/sui-framework/packages/move-stdlib", rev = "framework/mainnet" }
 
 [dev-addresses] # For use when developing this module
 address_to_be_filled_in = "0x101010101"
 ```
 
-Most of the sections in the package manifest are self explanatory, but named addresses can be a bit
-difficult to understand so it's worth examining them in a bit more detail.
+Most of the sections in the package manifest are self explanatory, but named
+addresses can be a bit difficult to understand so we examine them in more
+detail in [Named Addresses During
+Compilation](#named-addresses-during-compilation), but before that we'll first
+take a look at the `Move.lock` file and what it contains.
+
+## Move.lock
+
+The `Move.lock` file is generated at the root of the Move pacakge when the
+package is built. The `Move.lock` file contains information about your package
+and its build configuration, and acts as a communication layer between the Move
+compiler and other tools, like chain-specific command line interfaces and
+third-party package managers. 
+
+Like the `Move.toml` file, the `Move.lock` file is a text-based TOML file.
+Unlike the package manifest however, the `Move.lock` file is not intended for
+you to edit directly. Processes on the toolchain, like the Move compiler,
+access and edit the file to read and append relevant information to it. You
+also must not move the file from the root, as it needs to be at the same level
+as the `Move.toml` manifest in the package.
+
+If you are using source control for your package, make sure the `Move.lock` file
+is checked in to your repository. This ensures every build of your package is
+an exact replica of the original, and that changes to the build will be
+apparent as changes to the `Move.lock` file.
```

Comment:
I think this is generally fine. Some nitty gritty open thoughts on source control for `Move.lock`--

- I view it as "recommended practice" to check it in. Until we have automated address management, and even after that, I'm not 100% convinced it always makes sense to check in a changed `Move.lock`. E.g., maybe someone tries intentionally compiling with an old `sui` version and that updates the `Move.lock`, but they don't really want to check that in. I'm not sure exactly what I'd say here, but it's along the lines of "It's recommended practice to check in the `Move.lock` file that corresponds with your desired built or published package".

- It may very well be that folks just don't check in `Move.lock` because of reasons `hand-wave`. In this case, we (and tools that want to use `Move.lock` info) could be designed as best-effort. Concrete example: compiling a dependency with an `edition` could be looked up in the manifest if it's not in the `Move.lock` for some reason. Here I'm just trying to address the user who questions "What if I don't check in my Move.lock or don't want to?" and the TL;DR is "it helps tooling but things ought to ~mostly continue to work.

Reply:
>  I'm not sure exactly what I'd say here, but it's along the lines of "It's recommended practice to check in the Move.lock file that corresponds with your desired built or published package".

Yea totally agreed. I think one of the useful things about checking in the lock file is also serves as a canary for reviewers that something might have changed in a way that's not immediately apparent (i.e., a new dep got added by a dependency, or something like that). 

I'm generally trying to be fairly opinionated in the docs, so I think I will basically rip-off your "it's recommended practice.." and leave it at that (and if folks have reasons not to check it they don't _have_ to, but I think it's a good "default thing that you should do").

Reply:
Sounds good!